src/TrapException.cs

```diff
@@ -123,7 +184,12 @@ internal static TrapException FromOwnedTrap(IntPtr trap)
 
                 Native.wasm_trap_delete(trap);
 
-                return new TrapException(message, trapFrames);
+                var trappedException = new TrapException(message, trapFrames);
+                if (trappedExit)
+                    trappedException.ExitCode = exitStatus;
```

Comment:
This project tries to always use braces for conditional expressions to make it clear which lines are part of the conditional without having to rely on indentation.

```suggestion
                if (trappedExit)
                {
                    trappedException.ExitCode = exitStatus;
                }
```